### PR TITLE
Update to .NET 8 on Windows

### PR DESCRIPTION
**Pull request by bording**

### CODE REVIEW

File: src/ServiceControl/MessageFailures/Api/ArchiveMessagesController.cs [hunk 1]
```diff
@@ -2,17 +2,14 @@ namespace ServiceControl.MessageFailures.Api
...
     using InternalMessages;
     using NServiceBus;
-    using Infrastructure.WebApi;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Net.Http.Headers;
     using ServiceControl.Persistence;
-    using ServiceControl.Persistence.Infrastructure;
 
-    class ArchiveMessagesController : ApiController
+    class ArchiveMessagesController : Controller
```
Comment (awright18) [inline src/ServiceControl/MessageFailures/Api/ArchiveMessagesController.cs:12]: Maybe inherit from ControllerBase and [ApiController] attribute. Controller adds things like ViewBag and ViewResult which are unnecessary for API endpoints. [[ApiController](https://learn.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-8.0#apicontroller-attribute)] adds a bunch of nice framework goodies that can be useful with APIs.  (The attribute can be added at the method, class or assembly level. (copy paste this comment for other controllers.

File: src/Directory.Packages.props [hunk 2]
```diff
@@ -10,24 +10,18 @@
...
-    <PackageVersion Include="Microsoft.AspNet.SignalR.Client" Version="2.4.3" />
-    <PackageVersion Include="Microsoft.AspNet.WebApi" Version="5.3.0" />
-    <PackageVersion Include="Microsoft.AspNet.WebApi.OwinSelfHost" Version="5.3.0" />
     <PackageVersion Include="Microsoft.Data.SqlClient" Version="5.1.2" />
-    <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />
-    <PackageVersion Include="Microsoft.Extensions.Hosting" Version="7.0.1" />
-    <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="7.0.0" />
-    <PackageVersion Include="Microsoft.Extensions.Hosting.WindowsServices" Version="7.0.1" />
+    <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
+    <PackageVersion Include="Microsoft.Extensions.Hosting" Version="8.0.0" />
```
Comment (RoderickIveans) [inline src/Directory.Packages.props:20]: `Microsoft.Extensions.Hosting`、`NServiceBus.Metrics`、`NServiceBus.Metrics.ServiceControl.Msmq`  unused

Comment (bording) [reply src/Directory.Packages.props:20]: Thanks, but that's not true. We won't be removing those.

File: src/Custom.Build.props [hunk 3]
```diff
@@ -10,6 +10,7 @@
     <PowerShellModuleArtifactsPath>$(ArtifactsPath)PowerShellModules\$(PowerShellModuleName)\</PowerShellModuleArtifactsPath>
     <!-- Remove after upgrading to NServiceBus 8.x -->
     <ParticularAnalyzersVersion>0.9.0</ParticularAnalyzersVersion>
```
[the remark this answers is not in the document]
Comment (bording) [reply src/Custom.Build.props:12]: It does seem like we should try and do this, but it's going to generate a LOT of things we have to fix. I already spiked it, and it's going to be a lot of work.

[the remark this answers is not in the document]
Comment (mauroservienti) [reply src/Custom.Build.props:12]: Sooner or later, we have to do it anyway.

**Review by danielmarbach — APPROVED** [review]

**Pull request merged**